nautobot/ui/src/utils/api.js

```diff
@@ -18,31 +17,53 @@ import {
   in React and Redux's sites.
 */
 
-// eslint-disable-next-line no-unused-vars
-const staggeredBaseQuery = retry(fetchBaseQuery({ baseUrl: "" }), {
-    maxRetries: 5,
-}); // TODO: Make this smarter on which conditions it retries on
+/**
+ * A custom baseQuery that will retry requests that fail with a 5xx error code.
+ *
+ * @param {Object} args - The arguments passed to the query.
+ * @param {Object} api - The API object.
+ * @param {Object} extraOptions - Extra options passed to the query.
+ * @returns {Object} - The result of the query.
+ */
+const smartRetryBaseQuery = retry(
+    async (args, api, extraOptions) => {
+        const result = await fetchBaseQuery({ baseUrl: "" })(
+            args,
+            api,
+            extraOptions
+        );
+        const dontRetryHeaders = [400, 401, 403, 404, 405, 406, 407];
```

Comment:
no 500?

Reply:
Nope, 500 might be successful with a retry later. The backend could be restarted.